### PR TITLE
RNMT-2811 Merge native implementation

### PR DESCRIPTION
**Pull request by Chuckytuh**
As the name of the PR suggests, this PR basically adds a native implementation to the plugin for Android. 

Currently the branch [outsystems-native-implementation](https://github.com/OutSystems/cordova-plugin-geolocation/tree/outsystems-native-implementation) is being referenced by a customer on forge component so it should not be deleted for now.

### CODE REVIEW

File: plugin.xml [hunk 1]
```diff
@@ -38,6 +38,8 @@ xmlns:android="http://schemas.android.com/apk/res/android"
     <!-- android -->
     <platform name="android">
 
+        <framework src="com.google.android.gms:play-services-location:16.0.0" />
```
Comment (cmfsotelo) [inline plugin.xml:41]: Just a simple comment, we are introducing a dependency in here, that might clash with other plugins in other clients.
Many other libraries have dependencies on the `com.google.android.gms:play-services-base` that is a dependency of this framework here. Just a heads up

**Review by cmfsotelo — APPROVED** [review]

File: plugin.xml [hunk 2]
```diff
@@ -51,14 +53,25 @@ xmlns:android="http://schemas.android.com/apk/res/android"
...
         <source-file src="src/android/Geolocation.java" target-dir="src/org/apache/cordova/geolocation/" />
+        <source-file src="src/android/LocationContext.java" target-dir="src/org/apache/cordova/geolocation/" />
+        <source-file src="src/android/LocationUtils.java" target-dir="src/org/apache/cordova/geolocation/" />
+        <source-file src="src/android/OnLocationResultEventListener.java" target-dir="src/org/apache/cordova/geolocation/" />
+        <source-file src="src/android/LocationError.java" target-dir="src/org/apache/cordova/geolocation/" />
 
-        <js-module src="www/android/geolocation.js" name="geolocation">
-            <clobbers target="navigator.geolocation" />
+        <js-module src="www/Coordinates.js" name="Coordinates">
+            <clobbers target="Coordinates" />
```
Comment (usernuno) [inline plugin.xml:62]: Are all these clobbers required?

**Review by usernuno — APPROVED** [review]

File: plugin.xml (hunk 1 above)
Comment (Chuckytuh) [reply plugin.xml:41]: That's a good point and I'm not sure if this has been evaluated before when this was released to the customer experiencing problems. @pmaroco do you have any info on this?

File: plugin.xml (hunk 2 above)
Comment (Chuckytuh) [reply plugin.xml:62]: They exist on iOS and it seems it's a design decision from the original author in order to expose those objects to anyone willing to use them (see [here](https://github.com/OutSystems/cordova-plugin-geolocation/blob/outsystems/README.md#objects-read-only))

**Pull request merged**